.travis.yml

```diff
@@ -16,7 +16,11 @@ cache: pip
 # Avoid pip log from affecting cache
 before_cache: rm -fv ~/.cache/pip/log/debug.log
 
-# Install defaults to "pip install -r requirements.txt"
+install:
+  # Base requirements
+  - pip install -r requirements.txt
+  # Additional requirements for the unit and coverage tests
+  - pip install -r requirements-test.txt
```

Comment:
Does this work due to the way `-r` handles `requirements-test.txt`?
Could we have `pip install -r requirements-test.txt -c constraints.txt` in the install section of .travis.yml instead?

Reply:
Yes, it does (for a start the build passes).

The point of linking the constraints file from inside the requirements file is to keep everything in one place and ensure that it isn't missed. Also, you only need to look at the requirements file and not the Travis file as well to figure out where it's used.

Reply:
fair enough